91: Fix build status. Update GCC instructions.

JamesOrson opened the pull request:
Closes #91

brandonapol reviewed (approved):
Fixed URL for CI/CD and moved location of Jenkins for integration. Specified GCC compiler instead of Clang.

JamesOrson commented:
> Why is the CI/CD moved in the file? Also may want to specify about why GCC is used rather than Clang as it's holding Mac users back from using Daybreak with default Clang compiler

I moved the CI/CD build status because I thought it makes more sense in a different spot. If it is ugly, we can move it again. Just trying it out.

On Mac, you actually can use daybreak just fine with `clang` as well as Mac's `gcc` (that is actually just `clang`).
The flags that Daybreak uses internally work on both `gcc` and `clang` and should yield generally the same behavior for now.

The issue I am addressing in the doc is more an issue with Windows's default `cc` compiler.

Daybreak references `cc` rather than `gcc` or `clang` explicitly, which means on windows `cc` is generally referencing `msvc` or `cl` and oh boy those are nightmares. So on windows, you have to take special care to install `gcc`.

I have added a new issue to track that setup docs need some work for all platforms: https://github.com/ExoKomodo/Daybreak/issues/96

brandonapol commented:
Is there perhaps a way to have the language default to using gcc?
